CMakeLists.txt

```diff
@@ -1,33 +1,86 @@
 # CMake version >= 3.8.2
 # This requirement is for the "cxx_std_11" variable in target_compile_features().
 cmake_minimum_required(VERSION 3.8.2)
-project(libmorton CXX)
+project(libmorton VERSION 0.2.6)
 
-option(LIBMORTON_BUILD_TESTS "Build unit tests for libmorton" ON)
+option(BUILD_TESTING "Build unit tests for libmorton" ON)
+
+set(libmorton_INCLUDE_DIR "${libmorton_SOURCE_DIR}/libmorton/include")
+
+include(GNUInstallDirs)
+set(libmorton_INSTALL_PREFIX "${CMAKE_INSTALL_PREFIX}")
+set(libmorton_INSTALL_INCLUDE_DIR "${CMAKE_INSTALL_INCLUDEDIR}")
+set(libmorton_INSTALL_CMAKE_DIR "${CMAKE_INSTALL_DATADIR}/cmake/libmorton")
+set(libmorton_INSTALL_PKGCONFIG_DIR "${CMAKE_INSTALL_DATADIR}/pkgconfig")
 
 add_library(libmorton INTERFACE)
-target_compile_options(libmorton
-  INTERFACE
-    # gcc and clang 
-    $<$<OR:$<CXX_COMPILER_ID:GNU>,$<CXX_COMPILER_ID:Clang>>:-Wall -Wextra -Wpedantic>
-    # Intel compiler
-    $<$<CXX_COMPILER_ID:Intel>:$<IF:$<PLATFORM_ID:Windows>,/W3,-w3>>
-    # MSVC
-    $<$<CXX_COMPILER_ID:MSVC>:/W3>
-  )
-target_compile_definitions(libmorton
-  INTERFACE
-    # MSVC
-    # To avoid errors related to std::min/std::max
-    $<$<CXX_COMPILER_ID:MSVC>:NOMINMAX>
-  )
+
 target_include_directories(libmorton
   INTERFACE
-    libmorton/include
+    $<BUILD_INTERFACE:${libmorton_INCLUDE_DIR}>
+    $<INSTALL_INTERFACE:${libmorton_INSTALL_INCLUDE_DIR}>
   )
+
+# target_compile_definitions(libmorton
+#   INTERFACE
+#     # MSVC, to avoid errors related to std::min/std::max
+#     $<$<CXX_COMPILER_ID:MSVC>:NOMINMAX>
+#   )
+
 add_subdirectory(libmorton)
+
 add_library(libmorton::libmorton ALIAS libmorton)
 
-if (LIBMORTON_BUILD_TESTS)
-    add_subdirectory(test)
-endif()
\ No newline at end of file
+if(BUILD_TESTING)
+  enable_testing()
+
+  add_subdirectory(test)
+endif()
+
+install(TARGETS libmorton
+  EXPORT
+    libmortonTargets
+  PUBLIC_HEADER
+    DESTINATION
+      "${libmorton_INSTALL_INCLUDE_DIR}/libmorton"
+  )
+
+export(TARGETS libmorton
+  NAMESPACE libmorton::
+  FILE
+    "${libmorton_BINARY_DIR}/libmortonTargets.cmake"
+  )
+export(PACKAGE libmorton)
+install(EXPORT libmortonTargets
+  NAMESPACE libmorton::
+  DESTINATION
+    "${libmorton_INSTALL_CMAKE_DIR}"
+  )
+
+include(CMakePackageConfigHelpers)
+configure_package_config_file(
+    "${libmorton_SOURCE_DIR}/data/libmortonConfig.cmake.in"
+    "${libmorton_BINARY_DIR}/libmortonConfig.cmake"
+    INSTALL_DESTINATION "${libmorton_INSTALL_CMAKE_DIR}"
+    PATH_VARS libmorton_INSTALL_INCLUDE_DIR)
+write_basic_package_version_file(
+    "${libmorton_BINARY_DIR}/libmortonConfigVersion.cmake"
+    VERSION ${libmorton_VERSION}
+    COMPATIBILITY SameMajorVersion
```

Comment:
`SameMajorVersion` is to be discussed I believe. Values are explained in [CMake documentation](https://cmake.org/cmake/help/latest/module/CMakePackageConfigHelpers.html#command:write_basic_package_version_file).